<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>854</width>
    <height>692</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>831</width>
      <height>571</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="timeSpinBox">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>620</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <number>1000</number>
    </property>
   </widget>
   <widget class="QPushButton" name="updatePushButton">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>620</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>update</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>620</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;time&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_2">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>620</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;wa&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="waDoubleSpinBox">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>620</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="wbDoubleSpinBox">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>620</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_3">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>620</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;wb&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_4">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>620</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.8pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;wh&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="whDoubleSpinBox">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>620</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>854</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionCamera_Controls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCamera_Controls">
   <property name="text">
    <string>Camera Controls</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
   <slots>
    <signal>sig_sendSceneItem(QTreeWidgetItem*)</signal>
    <signal>sig_sendVertex(QListWidgetItem*)</signal>
    <signal>sig_sendFace(QListWidgetItem*)</signal>
    <signal>sig_sendHalfEdge(QListWidgetItem*)</signal>
    <signal>sig_sendVertexX(double)</signal>
    <signal>sig_sendVertexY(double)</signal>
    <signal>sig_sendVertexZ(double)</signal>
    <signal>sig_updateVertexX(double)</signal>
    <signal>sig_updateVertexY(double)</signal>
    <signal>sig_updateVertexZ(double)</signal>
    <signal>sig_updateFaceR(double)</signal>
    <signal>sig_updateFaceG(double)</signal>
    <signal>sig_updateFaceB(double)</signal>
    <signal>sig_sendJoint(QTreeWidgetItem*)</signal>
    <slot>slot_skeletonButtonPressed()</slot>
    <slot>slot_meshButtonPressed()</slot>
    <slot>slot_bindAtIndex(int)</slot>
    <slot>slot_updateVertices()</slot>
    <slot>slot_transformVertices(int,float,float,float,float,float,float)</slot>
    <slot>slot_halfEdgeSelectionChanged(QListWidgetItem*)</slot>
    <slot>slot_nextEdgeButtonPressed()</slot>
    <slot>slot_symEdgeButtonPressed()</slot>
    <slot>slot_faceSelectionChanged(QListWidgetItem*)</slot>
    <slot>slot_vertexSelectionChanged(QListWidgetItem*)</slot>
    <slot>slot_receiveVertexTranslateX(double)</slot>
    <slot>slot_receiveVertexTranslateY(double)</slot>
    <slot>slot_receiveVertexTranslateZ(double)</slot>
    <slot>slot_selectionID(int)</slot>
    <slot>slot_faceSelectChanged(QListWidgetItem*)</slot>
    <slot>slot_receiveFaceColorR(double)</slot>
    <slot>slot_receiveFaceColorG(double)</slot>
    <slot>slot_addVertex()</slot>
    <slot>slot_Triangulate()</slot>
    <slot>slot_halfEdgeSelectChanged(QListWidgetItem*)</slot>
    <slot>slot_vertexSelectChanged(QListWidgetItem*)</slot>
    <slot>slot_receiveFaceColorB(double)</slot>
    <slot>slot_subdivision()</slot>
    <slot>slot_extrude()</slot>
    <slot>slot_loadSkeleton()</slot>
    <slot>slot_updateVertex()</slot>
    <slot>slot_bindMesh()</slot>
    <slot>slot_getItemName()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
